server.py

```diff
@@ -185,6 +185,22 @@ def do_get_response(mongo_id):
 
     return jsonify({}), 404
 
+
+@app.route('/queue', methods=['POST'])
+def do_queue():
+    mongo_id = request.get_data().decode('UTF8')
+    # check document exists with id
+    result = do_get_response(mongo_id)
+    if result.status_code != 200:
+        return client_error("Unable to find document with id: " + mongo_id)
```

Comment:
This will hide the `server_error` that can happen in `do_get_response`. Might be better to just return the result from `do_get_response`?